File: packages/flic/android/src/main/java/io/capawesome/capacitorjs/plugins/flic/classes/results/StartScanResult.java
package io.capawesome.capacitorjs.plugins.flic.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.flic.FlicHelper;
import io.capawesome.capacitorjs.plugins.flic.interfaces.Result;
import io.flic.flic2libandroid.Flic2Button;

public class StartScanResult implements Result {

    @NonNull
    private final Flic2Button button;

    public StartScanResult(@NonNull Flic2Button button) {
        this.button = button;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("button", FlicHelper.createButtonJSObject(button));
        return result;
    }
}

File: packages/flic/ios/Plugin/Classes/Results/StartScanResult.swift
import Foundation
import Capacitor
import flic2lib

@objc public class StartScanResult: NSObject, Result {
    let button: FLICButton

    init(button: FLICButton) {
        self.button = button
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["button"] = FlicHelper.createButtonJSObject(button)
        return result as AnyObject
    }
}
